src/compute/src/render/top_k.rs

```diff
@@ -553,4 +563,85 @@ pub mod monoids {
             false
         }
     }
+
+    /// A shared portion of a thread-local top-1 monoid implementation.
+    #[derive(Debug)]
+    pub struct Top1MonoidShared {
+        order_key: Vec<ColumnOrder>,
+        left: DatumVec,
+        right: DatumVec,
+    }
+
+    impl Top1MonoidShared {
+        pub(crate) fn new(order_key: Vec<ColumnOrder>) -> Self {
+            Self {
+                order_key,
+                left: DatumVec::new(),
+                right: DatumVec::new(),
+            }
+        }
+    }
+
+    /// A monoid containing a row and a shared pointer to a shared structure.
+    /// Only suitable for thread-local aggregations.
+    #[derive(Debug, Clone)]
+    pub struct Top1MonoidLocal {
+        pub(crate) row: Row,
+        pub(crate) shared: Rc<RefCell<Top1MonoidShared>>,
+    }
+
+    impl Top1MonoidLocal {
+        pub(crate) fn into_row(self) -> Row {
+            self.row
+        }
+    }
+
+    impl PartialEq for Top1MonoidLocal {
+        fn eq(&self, other: &Self) -> bool {
+            self.row.eq(&other.row)
+        }
+    }
+
+    impl Eq for Top1MonoidLocal {}
+
+    impl Hash for Top1MonoidLocal {
+        fn hash<H: Hasher>(&self, state: &mut H) {
+            self.row.hash(state);
+        }
+    }
+
+    impl Ord for Top1MonoidLocal {
+        fn cmp(&self, other: &Self) -> Ordering {
+            let Top1MonoidShared {
+                left,
+                right,
+                order_key,
+            } = &mut *self.shared.borrow_mut();
+            debug_assert_eq!(&*order_key, &other.shared.borrow().order_key);
```

Comment:
I think if you want you might be able to do `Rc::ptr_eq` on the `self.shared` and `other.shared`, which might be a more direct way to get to this.

Reply:
Yup, that's better.